src/emitter.hpp

```diff
@@ -34,31 +34,117 @@ using std::thread;
+ *    - Each request is sent in its thread.
+ * 5. Once sent it will process the results of all the requests sent and will remove all successfully sent events from the database
+ */
 class Emitter {
 public:
+  /**
+   * @brief HTTP method used to send events to Snowplow Collector.
+   */
   enum Method {
     POST,
     GET
   };
+
+  /**
+   * @brief HTTP protocol used to send events to Snowplow Collector.
+   */
   enum Protocol {
     HTTP,
     HTTPS
   };
 
+  /**
+   * @brief Construct a new Emitter object
+   * 
+   * The `db_name` can be any valid path on your host file system (that can be created with the current user).
+   * By default it will create the required files wherever the application is being run from.
+   * 
+   * @param uri The URI to send events to
+   * @param method The request type to use (GET or POST)
+   * @param protocol The protocol to use (http or https)
+   * @param send_limit The maximum amount of events to send at a time
+   * @param byte_limit_post The byte limit when sending a POST request
+   * @param byte_limit_get The byte limit when sending a GET request
+   * @param db_name Defines the path and file name of the database
+   */
   Emitter(const string & uri, Method method, Protocol protocol, int send_limit, 
     int byte_limit_post, int byte_limit_get, const string & db_name);
   ~Emitter();
 
+  /**
+   * @brief Starts a long running daemon thread for sending events. Triggered automatically by the tracker.
+   */
   virtual void start();
+
+  /**
+   * @brief Stops the long running daemon thread for sending events. Triggered automatically by the tracker.
+   */
   virtual void stop();
+
+  /**
+   * @brief Adds an event to the database for sending. Triggered by tracker.
+   * 
+   * @param payload Event payload
+   */
   virtual void add(Payload payload);
+
+  /**
+   * @brief Force send queued events.
+   */
   virtual void flush();
 
+  /**
+   * @brief Get broken down collector URL.
+   * 
+   * @return CrackedUrl 
+   */
   CrackedUrl get_cracked_url() { return m_url; }
+
+  /**
+   * @brief Get the HTTP method.
+   * 
+   * @return Method HTTP method used for sending events to Collector
+  */
   Method get_method() { return m_method; }
+
+  /**
+   * @brief Get the send limit.
+   * 
+   * @return unsigned int The maximum amount of events to send at a time
+   */
   unsigned int get_send_limit() { return m_send_limit; }
+
+  /**
+   * @brief Get the byte limit for GET.
+   * 
+   * @return unsigned int The byte limit when sending a GET request
+   */
   unsigned int get_byte_limit_get() { return m_byte_limit_get; }
+
+  /**
+   * @brief Get the byte limit for POST.
+   * 
+   * @return unsigned int The byte limit when sending a POST request
+   */
   unsigned int get_byte_limit_post() { return m_byte_limit_post; }
+
+  /**
+   * @brief Check if the Emitter is started.
+   * 
+   * @return true Emitter is running
+   * @return false Emitter is not running
```

Comment:
Not sure on other API docs. On Javadoc I think you can have just a single return tag.
Wondering if a simple `@return true if Emitter is running` would work the same.

Reply:
Good point, the `@return false ...` is unnecessary, I'll remove it.